hank-server/src/main/java/com/liveramp/hank/ring_group_conductor/RingGroupUpdateTransitionFunctionImpl.java

```diff
@@ -233,9 +239,26 @@ private void removeFromReplicasFullyServing(Map<Domain, Map<Integer, Set<Host>>>
     return result;
   }
 
-  private Integer computeNumReplicasFullyServingRelevantData(Map<Domain, Map<Integer, Set<Host>>> domainToPartitionToHostsFullyServing,
-                                                             Set<DomainAndVersion> domainVersions,
-                                                             Host host) throws IOException {
+  enum LiveReplicaStatus {
+    UNDER_REPLICATED,
+    REPLICATED,
+    OVER_REPLICATED;
+
+    public boolean isFullyReplicated() {
+      return this == REPLICATED || this == OVER_REPLICATED;
+    }
+
+    public static LiveReplicaStatus min(LiveReplicaStatus... statuses) {
```

Comment:
Did we Java 8 this? If so, `Comparator` is a functional interface (and a `FunctionalInterface`) so we can do

    Collections.min(statuses, (lhs, rhs) -> lhs.ordinal() - rhs.ordinal())

or, if you find the arithmetic alarming,

    Collections.min(apps, (lhs, rhs) -> new CompareToBuilder().append(lhs.ordinal(), rhs.ordinal()).toComparison())